lib/puppet/application/device.rb

```diff
@@ -169,8 +174,15 @@ def main
     # find device list
     require 'puppet/util/network_device/config'
     devices = Puppet::Util::NetworkDevice::Config.devices
+    if options[:node]
+      devices.select! { |key, value| key == options[:node] }
```

Comment:
This feels kind of nasty to me, as it's a fairly major modification of global state. I assume access to the `Puppet::Util::NetworkDevice::Config.deivces` hash is hard-coded all over the place, making a cleaner solution a major pain?